### PR TITLE
Refactoring 

### PR DESCRIPTION
**Pull request by msdeshmukh009**
### Refactoring 
- Extracted hooks.
- Added search functionality.

### CODE REVIEW

File: src/components/modal/Modal.jsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { useState } from "react";
-import { usePlaylist } from "../../context";
+// import { usePlaylist } from "../../context";
```
Comment (BelideNikhil) [inline src/components/modal/Modal.jsx]: You can remove this comment

File: src/components/navbar/NavAside.jsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+          <li>
+            <NavLink
+              to="/watch-later"
+              className={({ isActive }) =>
+                isActive ? "nav-aside-link active-side-link" : "nav-aside-link"
+              }
+            >
+              Watch later
+            </NavLink>
+          </li>
```
Comment (BelideNikhil) [inline src/components/navbar/NavAside.jsx:65]: You can use map function here to reduce repeatability

File: src/components/navbar/navbar.css [hunk 3]
```diff
@@ -3,11 +3,25 @@
...
   grid-area: nav;
-  z-index: 1005;
+  z-index: 2;
   position: sticky;
   top: 0;
   background-color: var(--background-color);
 }
+.nav-search-bar {
+  border: 2px solid var(--secondary-color);
+  border-radius: 4.8px;
```
Comment (BelideNikhil) [inline src/components/navbar/navbar.css]: you should use rounded numbers  as decimals do no apply on pixels

File: src/components/navbar/navbar.css [hunk 4]
```diff
@@ -3,11 +3,25 @@
...
+}
+.nav-search-bar .form-field[type="search"] {
+  border: none;
+}
+.nav-search-bar .form-field:focus {
+  box-shadow: unset;
+}
 .nav {
   max-width: 2000px;
   margin: auto;
```
Comment (BelideNikhil) [inline src/components/navbar/navbar.css:27]: use width 100% here

File: src/reducers/videosReducer.js [hunk 5]
```diff
@@ -15,6 +15,9 @@ const videosReducer = (state, action) => {
 
     case SET_CATEGORY:
       return { ...state, selectedCategory: action.payload };
+
+    case "APPLY_SEARCH_TERM":
+      return { ...state, appliedSearchTerm: action.payload };
```
Comment (BelideNikhil) [inline src/reducers/videosReducer.js]: make this a variable as well

**Review by BelideNikhil — APPROVED** [review]
The code is very well written and nicely extracted
The UI is lit 🔥🔥 and works flawlessly

File: src/components/modal/Modal.jsx (hunk 1 above)
Comment (msdeshmukh009) [reply src/components/modal/Modal.jsx]: done

File: src/components/navbar/NavAside.jsx (hunk 2 above)
Comment (msdeshmukh009) [reply src/components/navbar/NavAside.jsx:65]: Ok, I will do that.

File: src/components/navbar/navbar.css (hunk 3 above)
Comment (msdeshmukh009) [reply src/components/navbar/navbar.css]: done

File: src/components/navbar/navbar.css (hunk 4 above)
Comment (msdeshmukh009) [reply src/components/navbar/navbar.css:27]: ok will do that

File: src/reducers/videosReducer.js (hunk 5 above)
Comment (msdeshmukh009) [reply src/reducers/videosReducer.js]: done

**Comment by pritam-kr** [comment]
Hey, Everything looks good to me.

- code is well written 
- The UI is also clean 
- all features are smoothly working good

**Review by pritam-kr — APPROVED** [review]

**Pull request merged**